### ISSUE TITLE
Add absolute_import to __init__ 

### ISSUE THREAD

**Issue by rbiswas4** [uwescience/shablona#27]
In one of my projects in python 2.7, I ran into a problem of distinguishing between a module in the package vs a package of the same name in the python path. I wanted to import an external package, but since I had a module in the package directory of the same name (that was a bad naming convention), 
it was importing that module.

```
from  module import MyClass # is module a module in the package or a package in the python path ?
```

The solution was to add the line

```
from __future__ import absolute_import
# Now the imports are clear
from module import MyClass # imports MyClass from package called module
from .module import MyPackageClass # import MyPackageClass form package module called module
```

This also allows you to import a module within package in the following way which I think is one of the recommended ways in python 3

```
from . import module
```

**Comment by arokem** [comment]
Looks like maybe this was sorted out on Python 2.7? https://www.python.org/dev/peps/pep-0328/

**Comment by rbiswas4** [comment]
@arokem 

Checkout a repository https://github.com/rbiswas4/quick_absimport_test which I made quickly for the purpose. You can install this using the setup file through the usual commands. The point is that this package contains a module [mynumpy.py](https://github.com/rbiswas4/quick_absimport_test/blob/master/test_imports/mynumpy.py)  which has a class called numpy. In [test.py](https://github.com/rbiswas4/quick_absimport_test/blob/master/test_imports/test.py) I want to write a function that prints out numpy.arange(0., 1., 0.1) in ways that it actually calls `numpy.arange(0., 1., 0.1)` or the module `tests_imports.numpy.arange` in clearly distinguishable ways. (Obviously, it would be stupid for someone to name a class by the well-known and highly used name of `numpy`, but this distinction is not so clear if the class in question is under development by multiple user in a multiple packages doing somewhat similar things.)

With the files in the repository which include the `from __future__ import absolute_import` on python 2.7
(Anaconda : Python 2.7.11 |Anaconda 2.1.0 (x86_64)| (default, Dec  6 2015, 18:57:58)) , 
I can do this correctly. Without the future statement, only the package numpy class works, the call to numpy ends up with a strange error:

# Using **future**

```
# to call the numpy we know
>>> import test_imports
>>> test_imports.check(0.)
x
[ 0.   0.1  0.2  0.3  0.4  0.5  0.6  0.7  0.8  0.9]
3.0
```

While if I replace the import [numpy](https://github.com/rbiswas4/quick_absimport_test/blob/master/test_imports/test.py#L3)
statement with (an uncommented version of) the previous [line](https://github.com/rbiswas4/quick_absimport_test/blob/master/test_imports/test.py#L2)
to call the package class numpy:

```
# Call the numpy class defined in *this package itself*

>>> import test_imports
>>> test_imports.check(0.)
x
HELP
(0.0, 1.0, 0.1)
3.0
>>>
```

This is all good.

If I get rid of the  `from __future__ import absolute_import`
statement at the top of [test.py](https://github.com/rbiswas4/quick_absimport_test/blob/master/test_imports/test.py), 

```
# Try calling the numpy we know
>>> import test_imports
>>> test_imports.check(0.)
x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rbiswas/.local/lib/python2.7/site-packages/test_imports/test.py", line 10, in check
    print(np.arange(0., 1., 0.1))
AttributeError: 'module' object has no attribute 'arange'
>>>
```

though the attempt to call the numpy defined within this package works correctly.

While there are probably other ways of obtaining the correct behavior, adding this absolute_import seems like a useful thing to do to minimize potential confusion.

**Comment by arokem** [comment]
I'm convinced. Where do we put this? In every `.py` file, or just in the `__init__.py` file?

**Comment by QuLogic** [comment]
You should put it in all files so that 2 and 3 are consistent. You can also do that for `division` and `print_function`, so that no-one will accidentally add old behaviour/syntax and break Python 3.

**Comment by arokem** [comment]
#41

**Comment by arokem** [comment]
Probably resolved through #41, but feel free to reopen if I misunderstand.
